.github/workflows/clang-tidy.yml

```diff
@@ -0,0 +1,27 @@
+name: tidy-modern
+
+on: [push, pull_request]
+
+jobs:
+  tidy:
+    runs-on: ubuntu-latest
+    container:
+      image: pointcloudlibrary/env:22.04
+    
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Run clang-tidy
+      run: |
+          wget https://apt.llvm.org/llvm.sh
+          chmod +x llvm.sh
+          ./llvm.sh 14
+          apt-get update
+          apt-get install -y clang-tidy-14 clang-tools-14 python3-pip
+          pip install pyyaml
+          cmake .
```

Comment:
I think you have to tell cmake explicitly to use clang, otherwise it defaults to gcc. This leads to the omp.h-not-found-errors, but might also lead to other errors

Reply:
AFAIK, the `cmake .` compiles `compile_commands.json`, which should contain the correct compiler flags.

Also, this:
[>Clang-tidy](https://developers.redhat.com/blog/2021/04/06/get-started-with-clang-tidy-in-red-hat-enterprise-linux) uses the same front-end libraries as the [Clang C language compiler](https://clang.llvm.org/). However, because it only takes source files as input, you can use clang-tidy for any C or C++ codebase no matter what compiler you are using.

Reply:
> AFAIK, the `cmake .` creates `compile_commands.json`, which should contain the correct compiler flags.

The correct compiler flags for the compiler chosen by cmake, right? So far that was gcc, so `compile_commands.json` contains the compiler flags needed by gcc. I guess clang-tidy needs the compiler flags for clang? That would include [something for omp, probably `-fopenmp`](https://stackoverflow.com/questions/33400462/omp-h-file-not-found-when-compiling-using-clang)

> Also, [this](https://developers.redhat.com/blog/2021/04/06/get-started-with-clang-tidy-in-red-hat-enterprise-linux):
> 
> > Clang-tidy uses the same front-end libraries as the [Clang C language compiler](https://clang.llvm.org/). However, because it only takes source files as input, you can use clang-tidy for any C or C++ codebase no matter what compiler you are using.

I would read that as "You can use clang-tidy even if you normally compile with gcc or another compiler", but in my understanding that doesn't contradict that clang-tidy needs flags for clang (of course this only matters when gcc and clang need different flags, which is the case in PCL, e.g. because of openmp)

BTW you currently don't tell clang-tidy where the `compile_commands.json` is located, or that it should search for the file at all. Does it still use `compile_commands.json`? The logs don't mention anything ...

Reply:
> BTW you currently don't tell clang-tidy where the `compile_commands.json` is located, or that it should search for the file at all. Does it still use `compile_commands.json`? The logs don't mention anything ...

It's baked into the `run-clang-tidy` script:
```
db_path = 'compile_commands.json'
```

Reply:
> It's baked into the `run-clang-tidy` script:

Ah, I see, thanks!

Please try telling cmake to use the clang compiler (e.g. like `CC=clang-14 CXX=clang++-14 cmake .`). For me locally at least, that fixed the omp.h-not-found errors.